### PR TITLE
Enhanced Registration and Login Flow for new users

### PR DESCRIPTION
**Pull request by bryan-robitaille**
# Summary | Résumé
 Enhances the user experience when registering to a more intuitive flow as described in the Figma designs.

Previous:
Registration => Confirmation => Login => Acceptable Use => MyForms
New:
Registration => Confirmation => Acceptable Use => New Account Info => MyForms

https://user-images.githubusercontent.com/25329319/202743999-1f1f20c3-6e07-43d5-9027-4f7c8e913cd9.mov

Previous:
Login => Confirmation => Login => Acceptable Use => MyForms
New:
Login => Confirmation => Acceptable Use => New Account Info => MyForms

https://user-images.githubusercontent.com/25329319/202744369-b1cc0ec3-1b9f-4e50-9eab-3b19959466bc.mov

There is also a small enhancement for when a bad password is entered during the confirmation flow.  The errors will be initially set on the screen when returning to the login to correct the error.

https://user-images.githubusercontent.com/25329319/202744750-4e1405e3-82f7-4311-906c-0df050662c84.mov

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

Ideally given more time I would like to refactor the hook and state management pattern used.  This is something we could look at as part of a future refactor.

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [x] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: pages/auth/login.tsx [hunk 1]
```diff
@@ -12,10 +12,13 @@ import { Confirmation } from "@components/auth/Confirmation/Confirmation";
 import UserNavLayout from "@components/globals/layouts/UserNavLayout";
 import * as Yup from "yup";
 
-const Register = () => {
-  const { username, cognitoError, setCognitoError, login } = useAuth();
+const Login = () => {
+  const { cognitoError, setCognitoError, login } = useAuth();
   const { t } = useTranslation(["login", "common"]);
```
Comment (thiessenp-cds) [inline pages/auth/login.tsx:17]: Looks like strings "UsernameOrPasswordIncorrect" is in congnito-errors, may need the namespace below

Comment (thiessenp-cds) [reply pages/auth/login.tsx:17]: But not worth stopping a PR over IMO

**Review by thiessenp-cds — APPROVED** [review]
LGTM
(Future todo automated tests :)

File: pages/auth/login.tsx (hunk 1 above)
Comment (craigzour) [reply pages/auth/login.tsx:17]: Nice catch! It was still working when I tested the branch though

Comment (bryan-robitaille) [reply pages/auth/login.tsx:17]: Thanks for the catch.  Adding in the missing namespace.

**Review by craigzour — DISMISSED** [review]
Great work!

File: lib/hooks/useAuth.ts [hunk 2]
```diff
@@ -150,7 +194,17 @@ export const useAuth = () => {
...
-    { username, password }: { username: string; password: string },
+    {
+      username,
+      password,
+      needsConfirmation,
+      didConfirm,
+    }: {
+      username: string;
+      password: string;
+      needsConfirmation: (s: boolean) => void;
```
Comment (Moro-Code) [inline lib/hooks/useAuth.ts:205]: You can use the following type `React.Dispatch<React.SetStateAction<boolean>>` This way you don't need to pass in an arrow function and can pass the state setter directly

**Review by Moro-Code — COMMENTED** [review]
Good to go. Minor comment if you can address before merging

File: lib/hooks/useAuth.ts (hunk 2 above)
Comment (bryan-robitaille) [reply lib/hooks/useAuth.ts:205]: Will do

**Pull request merged**